File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlFieldExpr.cs
namespace T1.SqlSharp.Expressions;

public class SqlFieldExpr : ISqlExpression
{
    public SqlType SqlType => SqlType.Field;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_FieldExpr(this);
    }

    public string FieldName { get; set; } = string.Empty;
    public string ToSql()
    {
        return FieldName;
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlFieldExpr.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlFieldExpr implements ISqlExpression {
    SqlType: SqlType = SqlType.Field;
    Span: TextSpan = new TextSpan();
    FieldName: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_FieldExpr(this);
    }

    ToSql(): string {
        return this.FieldName;
    }
} 